### PR TITLE
Fix CI tests with minimal dependencies and make dependency resolution more robust

### PR DESCRIPTION
**Pull request by lagru**
## Description

NumPy 2 is minimal dep for build but not for default. So installing defaults before building actually downgrades NumPy for the build step. This becomes a problem later when testing.

Still investigating how best to solve this.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpydoc/blob/main/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
...
```

### CODE REVIEW

**Comment by lagru** [comment]
@scikit-image/core this fixes our CI, hope we can get this merged. I also did some cleanup and made our dependency resolution more robust by carrying over requirements on successive pip install invocations.

File: tools/github/script.sh [hunk 1]
```diff
@@ -2,28 +2,36 @@
 # Fail on non-zero exit and echo the commands
 set -evx
 
-python -m pip install $PIP_FLAGS -r requirements/test.txt
+TEST_ARGS="--doctest-plus --cov=skimage --showlocals"
 
 
+# Combine requirement files for a more robust pip solve
+# installing successively may update previously constraint dependencies
```
Comment (mkcor) [inline tools/github/script.sh:9]:
```suggestion
# installing successively may update previously constrained dependencies
```

**Review by mkcor — APPROVED** [review]
Thank you, @lagru! Also for adding informative comments. :pray:

**Comment by mkcor** [comment]
The CI failure can't be related to 2517ada9739973bef. I'll merge before the end of today (my timezone, admittedly!) because this fix is pretty important for the overall development experience.

File: tools/github/before_install.sh [hunk 2]
```diff
@@ -3,48 +3,18 @@ set -ex
...
 python -m pip install $PIP_FLAGS -r requirements/build.txt
-# Default requirements are necessary to build because of lazy importing
-# They can be moved after the build step if #3158 is accepted
-python -m pip install $PIP_FLAGS -r requirements/default.txt
 
 # Show what's installed
 python -m pip list
 
-section () {
-    tools/header.py $1
```
Comment (soupault) [inline tools/github/before_install.sh:44]: I believe this change is not intended. Otherwise, the PR looks good. 👍🏻

Comment (mkcor) [reply tools/github/before_install.sh:44]: Oh no! It took me a few minutes to squash and merge (proofreading the long-form commit messages) and I missed this comment... :(

**Comment by lagru** [comment]
That change was actually intended. As far as I'm aware `section ()` isn't used anywhere? So it's the cleanup part of the commit description.

**Comment by soupault** [comment]
👍🏻 Seems, it used to be called in Travis builds - https://github.com/scikit-image/scikit-image/commit/85711410b056291742853e70417931fd6ebf83ee .
I would suggest removing `header.py` as redundant then.

**Pull request merged**